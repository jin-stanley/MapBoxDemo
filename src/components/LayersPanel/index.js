import React from "react";
import { makeStyles } from "@material-ui/core/styles";

import SettingWrapper from "../SettingWrapper";

const LayersPanel = (props) => {
  const classes = useStyles();

  // Here Using mock data for now, we can set real data by api
  const layerStyles = [
    {
      id: 1,
      name: "Satellite View",
      url: "mapbox://styles/stanleyjin/ckf4whmdl21w519qhwpvbmruo",
      img:
        "https://upload.wikimedia.org/wikipedia/commons/c/c3/%E9%95%87%E5%B7%B4_%E5%8D%AB%E6%98%9F%E5%9B%BE.JPG",
    },
    {
      id: 2,
      name: "Simple View",
      url: "mapbox://styles/stanleyjin/ckf4wz3jn13wt19o4ha6xsp9q",
      img:
        "https://crmfile.alicdn.com/crmfile-open/wmqbbs/windid/1809/thread/309_2645085_d232b975e15394a.jpg",
    },
  ];

  return (
    <SettingWrapper name="Layers" handleCloseSetting={props.handleCloseSetting}>
      <div className={classes.container}>
        {layerStyles.map((layerStyle) => {
          return (
            <div
              key={layerStyle.id}
              className={classes.layerItem}
              onClick={() => props.handleChangeLayerStyle(layerStyle.url)}
            >
              <div
                className={classes.layerItemPic}
                style={{
                  backgroundImage: `url(${layerStyle.img})`,
                }}
              />
              <div
                className={classes.layerItemName}
                style={{
                  color: props.selectedLayer === layerStyle.url && "#1f4b5c",
                }}
              >
                {layerStyle.name}
                {props.selectedLayer === layerStyle.url && (
                  <span>
                    <br />
                    (Current)
                  </span>
                )}
              </div>
            </div>
          );
        })}
      </div>
    </SettingWrapper>
  );
};

const useStyles = makeStyles({
  container: {
    display: "flex",
    flexFlow: "wrap",
  },
  layerItem: {
    display: "flex",
    alignItems: "center",
    margin: "8px",
    cursor: "pointer",
    "&:hover": {
      color: "#7d3f13",
    },
  },
  layerItemPic: {
    width: "72px",
    height: "72px",
    backgroundSize: "cover",
    backgroundPosition: "right top",
    borderRadius: "8px",
  },
  layerItemName: {
    textAlign: "center",
    fontSize: "16px",
    fontWeight: 800,
    margin: "8px",
    "& span": {
      fontSize: "10px",
    },
  },
});

export default LayersPanel;
