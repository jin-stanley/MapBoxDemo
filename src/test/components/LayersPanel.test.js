import React from "react";
import renderer from "react-test-renderer";

import LayersPanel from "../../components/LayersPanel";

it("LayersPanel can render success", () => {
  const tree = renderer.create(<LayersPanel />).toJSON();
  expect(tree).toMatchSnapshot();
});
