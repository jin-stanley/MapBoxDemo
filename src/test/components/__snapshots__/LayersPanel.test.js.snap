// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayersPanel can render success 1`] = `
<div
  className="makeStyles-container-5"
>
  <div
    className="makeStyles-header-6"
  >
    <p>
      Layers
    </p>
    <svg
      className="makeStyles-closeIcon-8"
      onClick={[Function]}
    >
      x.svg
    </svg>
  </div>
  <div
    className="makeStyles-content-7"
  >
    <div
      className="makeStyles-container-1"
    >
      <div
        className="makeStyles-layerItem-2"
        onClick={[Function]}
      >
        <div
          className="makeStyles-layerItemPic-3"
          style={
            Object {
              "backgroundImage": "url(https://upload.wikimedia.org/wikipedia/commons/c/c3/%E9%95%87%E5%B7%B4_%E5%8D%AB%E6%98%9F%E5%9B%BE.JPG)",
            }
          }
        />
        <div
          className="makeStyles-layerItemName-4"
          style={
            Object {
              "color": false,
            }
          }
        >
          Satellite View
        </div>
      </div>
      <div
        className="makeStyles-layerItem-2"
        onClick={[Function]}
      >
        <div
          className="makeStyles-layerItemPic-3"
          style={
            Object {
              "backgroundImage": "url(https://crmfile.alicdn.com/crmfile-open/wmqbbs/windid/1809/thread/309_2645085_d232b975e15394a.jpg)",
            }
          }
        />
        <div
          className="makeStyles-layerItemName-4"
          style={
            Object {
              "color": false,
            }
          }
        >
          Simple View
        </div>
      </div>
    </div>
  </div>
</div>
`;
